mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/store/MantisStorageProvider.java

```diff
@@ -183,4 +183,12 @@ List<MantisWorkerMetadataWritable> getArchivedWorkers(String jobid)
     void deleteExpiredDisableTaskExecutorRequest(DisableTaskExecutorsRequest request) throws IOException;
 
     List<DisableTaskExecutorsRequest> loadAllDisableTaskExecutorsRequests(ClusterID clusterID) throws IOException;
+
+    void storeNewJobArtifact(JobArtifact jobArtifact) throws JobArtifactIDAlreadyExistsException, IOException;
+
+    List<JobArtifact> initJobArtifacts();
```

Comment:
what's this one for? Do we need to load whole list in master memory?

Reply:
in theory yes that was the original idea but now I'm not using it anymore so maybe I should get rid of it.

Reply:
@fdc-ntflx As part of PR #285, I'm merging the functionality in these two classes together. It'd save me some post-merge migration if you could resubmit this diff on top of that PR. Probably saves you some testing pain as well hopefully!